### PR TITLE
pygame.font.get_ttf_version() request from issue 3087

### PR DESCRIPTION
**Pull request by mariospapaz**
issue link: https://github.com/pygame/pygame/issues/3087

Now the user can get the SDL_ttf version in both linked and compiled ways.

Documentation:
Chapter 3.1.1
https://www.libsdl.org/projects/docs/SDL_ttf/SDL_ttf.html#SEC7

Screen Shot 2022-03-27 at 9 46 50 PM

### CODE REVIEW

**Comment by novialriptide** [comment]
Looks good, all you need to do now is add this to the docs.

File: src_c/font.c [hunk 1]
```diff
@@ -788,14 +789,53 @@ get_default_font(PyObject *self, PyObject *_null)
...
+}
+
+
 static PyMethodDef _font_methods[] = {
     {"init", (PyCFunction)fontmodule_init, METH_NOARGS, DOC_PYGAMEFONTINIT},
     {"quit", (PyCFunction)fontmodule_quit, METH_NOARGS, DOC_PYGAMEFONTQUIT},
     {"get_init", (PyCFunction)get_init, METH_NOARGS, DOC_PYGAMEFONTGETINIT},
     {"get_default_font", (PyCFunction)get_default_font, METH_NOARGS,
      DOC_PYGAMEFONTGETDEFAULTFONT},
+    {"get_ttf_version", (PyCFunction) get_ttf_version, METH_VARARGS | METH_KEYWORDS, DOC_PYGAMEFONTGETINIT},
```
Comment (novialriptide) [inline src_c/font.c:834]: It would be better to rename this function to `get_version()`, since `freetype` uses `get_version()` as well.

**Review by novialriptide — CHANGES_REQUESTED** [review]

File: src_c/font.c (hunk 1 above)
Comment (Starbuck5) [reply src_c/font.c:834]: Freetype is the odd one out, all the rest use this format.

File: test/font_test.py [hunk 2]
```diff
@@ -35,6 +35,9 @@ def setUp(self):
 
     def tearDown(self):
         pygame_font.quit()
+    
+    def get_ttf_version(self):
+        pygame_font.get_ttf_version()
```
Comment (Starbuck5) [inline test/font_test.py:40]: I know you're not done with this yet, but the reason this doesn't run is that test methods must start with `test_`

Comment (mariospapaz) [reply test/font_test.py]: Issue resolved.

Which are the other steps to get it working with the docs ?

**Comment by mariospapaz** [comment]
if there are no further requests regarding this function, the feature is finally finished.

File: test/font_test.py [hunk 3]
```diff
@@ -36,6 +36,9 @@ def setUp(self):
     def tearDown(self):
         pygame_font.quit()
 
+    def test_get_sdl_ttf_version(self):
+        pygame_font.get_sdl_ttf_version()
```
Comment (novialriptide) [inline test/font_test.py:40]: Must add another test for unlinked as well

Comment (novialriptide) [inline test/font_test.py:40]: Might as well add a test if this function actually returns a string or tuple with 3 ints.

**Review by novialriptide — CHANGES_REQUESTED** [review]
Good job on the docs. Just a few more changes.

**Review by MyreMylar — CHANGES_REQUESTED** [review]
Given this a good local test and code wise it all seems to work 👍 

This is how the docs look right now:

image


I'm thinking we should add an extra sentence explaining what SDL_ttf is to the average pygame user, and a full stop after the first sentence.

Something like:

> Returns a tuple of integers that identify SDL_ttf's version. SDL_ttf is the underlying font rendering library, written in C, on which pygame's font module depends.

```

**Comment by MyreMylar** [comment]
Also, this test will always fail for ftfont because that doesn't use SDL_ttf.

So perhaps the test should be:

```
def test_get_sdl_ttf_version(self):
    if pygame_font.__name__ != "pygame.ftfont":
        pygame_font.get_sdl_ttf_version()
```

**Comment by mariospapaz** [comment]
Dear @MyreMylar, I've added the things you requested

An overview of the first request:
Screen Shot 2022-04-03 at 12 33 58 PM

**Review by MyreMylar — APPROVED** [review]

File: buildconfig/stubs/pygame/font.pyi [hunk 4]
```diff
@@ -11,6 +11,7 @@ UCS4: Literal[1]
 def init() -> None: ...
 def quit() -> None: ...
 def get_init() -> bool: ...
+def get_sdl_ttf_version() -> Tuple[int, int, int]: ...
```
Comment (ankith26) [inline buildconfig/stubs/pygame/font.pyi:14]:
```suggestion
def get_sdl_ttf_version(linked: bool = True) -> Tuple[int, int, int]: ...
```

File: docs/reST/ref/font.rst [hunk 5]
```diff
@@ -79,6 +79,19 @@ solves no longer exists, it will likely be removed in the future.
 
    .. ## pygame.font.get_default_font ##
 
+.. function:: get_sdl_ttf_version
+
+   | :sl:`gets SDL_ttf version`
+   | :sg:`get_sdl_ttf_version() -> (major, minor, patch)`
```
Comment (ankith26) [inline docs/reST/ref/font.rst:85]:
```suggestion
   | :sg:`get_sdl_ttf_version(linked=True) -> (major, minor, patch)`
```

File: docs/reST/ref/font.rst [hunk 6]
```diff
@@ -79,6 +79,19 @@ solves no longer exists, it will likely be removed in the future.
...
+
+   Returns a tuple of integers that identify SDL_ttf's version.
+   SDL_ttf is the underlying font rendering library, written in C,
+   on which pygame's font module depends.
+
```
Comment (ankith26) [inline docs/reST/ref/font.rst:90]: Needs a mention of the `linked` argument here

**Review by ankith26 — COMMENTED** [review]
Left some additional reviews in addition to the tests review I already gave on discord

**Review by MyreMylar — APPROVED** [review]
Looking good 👍

**Comment by MyreMylar** [comment]
lint test is failing on the `font_test.py`. Just needs a quick:

```
black test/display_test.py
```

to pass the test. Think it is just spacing around the new function in the test.

Everything looks good though

**Review by MyreMylar — COMMENTED** [review]
Couple of fixes for missing docs regeneration.

Apparently I can apply them myself, so I did.

**Review by ankith26 — APPROVED** [review]
LGTM! Thanks for the PR 🎉

**Comment by illume** [comment]
get_ttf_version was a bit confusingly named to me. The function is for getting SDL_ttf version, so probably a better name would be get_sdl_ttf_version()? We have pygame.mixer.get_sdl_mixer_version() and pygame.get_sdl_version().

**Comment by Starbuck5** [comment]
> get_ttf_version was a bit confusingly named to me. The function is for getting SDL_ttf version, so probably a better name would be get_sdl_ttf_version()? We have pygame.mixer.get_sdl_mixer_version() and pygame.get_sdl_version().

get_sdl_ttf_version is what was actually implemented, the PR was misnamed. I've fixed that.

**Comment by illume** [comment]
Ah, cool. Thanks for updating the title.

**Pull request merged**